### PR TITLE
PHP 8.4 support

### PR DESCRIPTION
**Pull request by phpfui**
* Removed a depreciation message
* Updated PHPUnit tests to not have any issues
* Added testing support for PHP 8.1 through 8.4

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "2.x-dev"
+            "dev-master": "3.x-dev"
```
Comment (alek13) [inline composer.json:38]: if we want only to support php8.4, we don't need to change version

File: tests/TestCase.php [hunk 2]
```diff
@@ -1,11 +0,0 @@
-<?php
-namespace Slack\Tests;
-
-use PHPUnit\Framework\TestCase as PhpUnitTestCase;
```
Comment (alek13) [inline tests/TestCase.php]: please, keep this class

File: src/BlockElement/AbstractStaticSelect.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
 namespace Maknz\Slack\BlockElement;
 
 use InvalidArgumentException;
-use Maknz\Slack\Object\OptionGroup;
+use Maknz\Slack\CompositionObject\OptionGroup;
```
Comment (alek13) [inline src/BlockElement/AbstractStaticSelect.php:5]: this breaks BC, for ex., if some one already extends these classes.
so, It's better to keep the old name.

File: tests/Block/ActionsUnitTest.php [hunk 4]
```diff
@@ -5,9 +5,8 @@
 use Maknz\Slack\Block\Actions;
 use Maknz\Slack\BlockElement\StaticSelect;
 use Maknz\Slack\BlockElement\Text;
-use Slack\Tests\TestCase;
 
-class ActionsUnitTest extends TestCase
+class ActionsUnitTest extends \PHPUnit\Framework\TestCase
```
Comment (alek13) [inline tests/Block/ActionsUnitTest.php]: these changes are unnecessary

**Review by alek13 — CHANGES_REQUESTED** [review]

File: src/BlockElement/AbstractStaticSelect.php (hunk 3 above)
Comment (alek13) [reply src/BlockElement/AbstractStaticSelect.php:5]: mmm... sorry, I already forget all of things.
Do you try to add php8.4 support to `2.x` branch ?
Is it possible ?

Comment (phpfui) [reply src/BlockElement/AbstractStaticSelect.php:5]: There seem to be some extraneous commits in here. Thought I branched off main. Will check.

File: composer.json (hunk 1 above)
Comment (phpfui) [reply composer.json:38]: I did not make this change, so no idea about this. It looks like I branched from master, so this was already there. You should correct it yourself to be correct.

File: tests/TestCase.php (hunk 2 above)
Comment (phpfui) [reply tests/TestCase.php]: Done

File: src/BlockElement/AbstractStaticSelect.php (hunk 3 above)
Comment (phpfui) [reply src/BlockElement/AbstractStaticSelect.php:5]: I did branch from master, so this was already there. Not sure why, but you should probably fix, as it is not my change.

File: tests/Block/ActionsUnitTest.php (hunk 4 above)
Comment (phpfui) [reply tests/Block/ActionsUnitTest.php]: I reverted all of this.  There is one warning message that TeestCase has no tests, but that can be ignored.

File: composer.json (hunk 1 above)
Comment (alek13) [reply composer.json:38]: you need to rebase on `2.x`

**Comment by phpfui** [comment]
Closing this. Not sure how I don't have the 2.x branch, but I don't seem to have it. I will open a new PR based on the last 2.x commit.

**Pull request closed without merging**